File: source/generic/s3eKochava.cpp
/*
Generic implementation of the s3eKochava extension.
This file should perform any platform-indepedentent functionality
(e.g. error checking) before calling platform-dependent implementations.
*/

/*
 * NOTE: This file was originally written by the extension builder, but will not
 * be overwritten (unless --force is specified) and is intended to be modified.
 */


#include "s3eKochava_internal.h"
s3eResult s3eKochavaInit()
{
    //Add any generic initialisation code here
    return s3eKochavaInit_platform();
}

void s3eKochavaTerminate()
{
    //Add any generic termination code here
    s3eKochavaTerminate_platform();
}

void s3eKochavaCreate(const char* appId)
{
	s3eKochavaCreate_platform(appId);
}

void s3eKochavaEvent(const char* eventName, const char* eventData)
{
	s3eKochavaEvent_platform(eventName, eventData);
}

void s3eKochavaEventSpatial(const char* eventName, int x, int y, int z, const char* eventData)
{
	s3eKochavaEventSpatial_platform(eventName, x, y, z, eventData);
}

void s3eKochavaDeepLinkEvent(const char* uri)
{
	s3eKochavaDeepLinkEvent_platform(uri);
}

void s3eKochavaEnableDebug(bool value)
{
	s3eKochavaEnableDebug_platform(value);
}

void s3eKochavaSetErrorDebug(bool value)
{
	s3eKochavaSetErrorDebug_platform(value);
}

File: source/android/s3eKochava.java
/*
java implementation of the s3eKochava extension.

Add android-specific functionality here.

These functions are called via JNI from native code.
*/
/*
 * NOTE: This file was originally written by the extension builder, but will not
 * be overwritten (unless --force is specified) and is intended to be modified.
 */
import com.ideaworks3d.marmalade.LoaderAPI;
// Comment in the following line if you want to use ResourceUtility
// import com.ideaworks3d.marmalade.ResourceUtility;

import com.ideaworks3d.marmalade.LoaderActivity;

import com.kochava.android.tracker.Feature;

class s3eKochava
{
	private Feature kTracker = null;
	
    public void s3eKochavaCreate(String appId)
    {
		java.util.HashMap<String, Object> dataMap = new java.util.HashMap<String, Object>();
		dataMap.put(Feature.INPUTITEMS.KOCHAVA_APP_ID, appId );
		kTracker = new Feature ( LoaderActivity.m_Activity, dataMap );
    }
    public void s3eKochavaEvent(String eventName, String eventData)
    {
		if (kTracker != null) {
			kTracker.event(eventName, eventData);
		}
    }
    public void s3eKochavaEventSpatial(String eventName, int x, int y, int z, String eventData)
    {
		if (kTracker != null) {
			kTracker.eventSpatial(eventName, x, y, z, eventData);
		}
    }
    public void s3eKochavaDeepLinkEvent(String uri)
    {
		if (kTracker != null) {
			kTracker.deepLinkEvent(uri);
		}
    }
    public void s3eKochavaEnableDebug(boolean value)
    {
		Feature.enableDebug(value);
    }
    public void s3eKochavaSetErrorDebug(boolean value)
    {
		Feature.setErrorDebug(value);
    }
}
